macaroon_service.go

```diff
@@ -367,3 +412,50 @@ func NewBoltMacaroonStore(dbPath, dbFileName string,
 
 	return rks, db, nil
 }
+
+// macaroonMatchesPermissions checks if the macaroon at the cfg.MacaroonPath
+// matches the required permissions. It returns true if the macaroon matches the
+// required permissions.
+func (ms *MacaroonService) macaroonMatchesPermissions() (bool, error) {
+	macBytes, err := os.ReadFile(ms.cfg.MacaroonPath)
+	if err != nil {
+		returnErr := fmt.Errorf("unable to read macaroon path: %v", err)
+		return false, returnErr
+	}
+
+	// Make sure it actually is a macaroon by parsing it.
+	oldMac := &macaroon.Macaroon{}
+	if err := oldMac.UnmarshalBinary(macBytes); err != nil {
+		return false, fmt.Errorf("unable to decode macaroon: %v", err)
+	}
+
+	macSlice := macaroon.Slice{oldMac}
+	authChecker := ms.Checker.Auth(macSlice)
+
+	ctx := context.Background()
```

Comment:
Would really appreciate some feedback on how to use contexts here:  
I still don't feel that I fully understand the best practices for how context(s) should be used. Is ok to create a new context here, or should this rather be done in the outer service that uses the lndclient, and then pass that context to the `MacaroonService` via the `Start` function?

Reply:
This should be fine as is cause I dont think `.Allow` is at all expected to hang. Yes it is fine to create a new context here. What you could do to be safe if you wanted is:

```
	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
	_, err = authChecker.Allow(ctx, requiredPerms...)
	cancel()
```
(and then just put the `time.Second * 5` part into a const).

Reply:
Thanks a lot for explanation and for the suggested safety fix! I've implemented it with the latest push :)